uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js

```diff
@@ -151,31 +151,31 @@ define([
     }
 
     /**
-     * Get profile URL for the given contact id
+     * Filters leave requests out by following rule:
+     *   If both TOIL and Non-TOIL (Leave/Sickness) requests exist
+     *   for the same day, remove Non-TOIL requests from the collection
      *
-     * @param {string/int} contactId
+     * @param  {Array} leaveRequests collection of leave requests
+     * @return {Array} collection of references to filtered leave requests
      */
-    function getContactUrl (contactId) {
-      return CRM.url('civicrm/contact/view', { cid: contactId });
+    function filterLeaveRequestsToShowInCell (leaveRequests) {
```

Comment:
Why do we need to filter the leave requests that we are going to display? Isn't this ruled by the API anyways? (ie: the API wouldn't allow you to violate these constraints)

Reply:
@reneolivo we need to show all leave requests in the tooltip, however, in the cell we have a special rule: if there are both non-toil and toil requests, then show only toil requests.